File: backends/credhub/src/test/java/org/cloudfoundry/credhub/helpers/FakeOauthTokenFilter.java
package org.cloudfoundry.credhub.helpers;

import java.io.IOException;
import java.util.regex.Pattern;

import javax.servlet.Filter;
import javax.servlet.FilterChain;
import javax.servlet.FilterConfig;
import javax.servlet.ServletException;
import javax.servlet.ServletRequest;
import javax.servlet.ServletResponse;
import javax.servlet.http.HttpServletRequest;

public class FakeOauthTokenFilter implements Filter {

  @Override
  public void init(FilterConfig filterConfig) throws ServletException {

  }

  @Override
  public void doFilter(ServletRequest request, ServletResponse response, FilterChain chain)
    throws IOException, ServletException {
    HttpServletRequest servletRequest = (HttpServletRequest) request;
    String header = servletRequest.getHeader("Authorization");

    if (header == null) {
      throw new ServletException("Missing Authorization header");
    }

    Pattern p = Pattern.compile("Bearer .+");
    if (!p.matcher(header).matches()) {
      throw new ServletException("Missing auth token in Authorization header");
    }
    chain.doFilter(request, response);
  }


  @Override
  public void destroy() {

  }
}

File: backends/credhub/src/test/kotlin/org/cloudfoundry/credhub/helpers/FakeOauthTokenFilter.kt
package org.cloudfoundry.credhub.helpers

import jakarta.servlet.Filter
import jakarta.servlet.FilterChain
import jakarta.servlet.FilterConfig
import jakarta.servlet.ServletException
import jakarta.servlet.ServletRequest
import jakarta.servlet.ServletResponse
import jakarta.servlet.http.HttpServletRequest
import java.io.IOException
import java.util.regex.Pattern

class FakeOauthTokenFilter : Filter {
    @Throws(ServletException::class)
    override fun init(filterConfig: FilterConfig) {
    }

    @Throws(IOException::class, ServletException::class)
    override fun doFilter(
        request: ServletRequest,
        response: ServletResponse,
        chain: FilterChain,
    ) {
        val servletRequest = request as HttpServletRequest
        val header = servletRequest.getHeader("Authorization") ?: throw ServletException("Missing Authorization header")
        val p = Pattern.compile("Bearer .+")
        if (!p.matcher(header).matches()) {
            throw ServletException("Missing auth token in Authorization header")
        }
        chain.doFilter(request, response)
    }

    override fun destroy() {}
}
